sdk/lib/collection/hash_map.dart

```diff
@@ -17,6 +17,7 @@ typedef _Equality<K> = bool Function(K a, K b);
 typedef _Hasher<K> = int Function(K object);
 
 /// A hash-table based implementation of [Map].
+/// The [HashMap] is unordered (the order of iteration is not guaranteed).
```

Comment:
I don't really feel having the explanation of unordered inside brackets. I would rather have it inside the sentence somehow. What @stepthomp thinks?